runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -6124,24 +6124,33 @@ done:;
 			 */
 			switch (romCPEntry->cpType >> J9DescriptionReturnTypeShift) {
 			case J9DescriptionReturnTypeBoolean:
-				value = (j9object_t)(UDATA)J9VMJAVALANGBOOLEAN_VALUE(_currentThread, value);
+				*(I_32 *)_sp = (I_32)J9VMJAVALANGBOOLEAN_VALUE(_currentThread, value);
 				break;
 			case J9DescriptionReturnTypeByte:
-				value = (j9object_t)(UDATA)J9VMJAVALANGBYTE_VALUE(_currentThread, value);
+				*(I_32 *)_sp = (I_32)J9VMJAVALANGBYTE_VALUE(_currentThread, value);
 				break;
 			case J9DescriptionReturnTypeChar:
-				value = (j9object_t)(UDATA)J9VMJAVALANGCHARACTER_VALUE(_currentThread, value);
+				*(I_32 *)_sp = (I_32)J9VMJAVALANGCHARACTER_VALUE(_currentThread, value);
 				break;
 			case J9DescriptionReturnTypeShort:
-				value = (j9object_t)(UDATA)J9VMJAVALANGSHORT_VALUE(_currentThread, value);
+				*(I_32 *)_sp = (I_32)J9VMJAVALANGSHORT_VALUE(_currentThread, value);
 				break;
 			case J9DescriptionReturnTypeFloat:
-				value = (j9object_t)(UDATA)J9VMJAVALANGFLOAT_VALUE(_currentThread, value);
+				*(I_32 *)_sp = (I_32)J9VMJAVALANGFLOAT_VALUE(_currentThread, value);
 				break;
 			case J9DescriptionReturnTypeInt:
-				value = (j9object_t)(UDATA)J9VMJAVALANGINTEGER_VALUE(_currentThread, value);
+				*(I_32 *)_sp = (I_32)J9VMJAVALANGINTEGER_VALUE(_currentThread, value);
+				break;
+			case J9DescriptionReturnTypeDouble: /* fall through */
+			case J9DescriptionReturnTypeLong:
+				/* double and long value cannot be loaded by ldc as they require 2 slots */
+				Assert_VM_unreachable();
+				break;
+			default:
+				*_sp = (UDATA)value;
```

Comment:
Why add the default when it duplicates the write below on line 6155?

Reply:
this avoid having goto for each of the switch case

Reply:
Fair enough.  The code will be ugly one or the other =)